### PR TITLE
Version 0.20.0

### LINKED ISSUE #1890: The `main` procmacro is not re-exported in the current published releases of `esp-hal` and `esp-hal-embassy`
There was an unfortunately oversight with the previous release of `esp-hal-embassy`, in that it does not re-export the `esp_hal_procmacros::main` macro. Since `esp-hal` no longer enables the `esp-hal-procmacros/embassy` feature it's also not exported from this package.

For the time being, workarounds are:

1.) Add `esp-hal-procmacros` as a direct dependency, with the `embassy` feature enabled, and import the macro from there
2.) Use the `main` branch of the repository for both packages, and the issue has been resolved in this branch

@MabezDev @bjoernQ we discussed possibly publishing an interim release of this package, we should discuss this during our meeting on Monday and come to a decision.

### LINKED ISSUE #1897: Latest release (v0.2.2) of `xtensa-lx-rt-proc-macros` not used by published `xtensa-lx-rt`
This due to the way the dependency is specificed; Rust treats it as a [multiple locations dependency](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#multiple-locations) and will use `version = "=0.2.1"` whenever this crate is pulled from [crates.io](https://crates.io)
```toml
macros = { version = "0.2.1", package = "xtensa-lx-rt-proc-macros", path = "./procmacros" }
```

When forcing the update using `cargo update xtensa-lx-rt-proc-macros --precise 0.2.2`, cargo fails with the following error:
```
    Updating crates.io index
error: failed to select a version for the requirement `xtensa-lx-rt-proc-macros = "=0.2.1"`
candidate versions found which didn't match: 0.2.2
location searched: crates.io index
required by package `xtensa-lx-rt v0.16.0`
    ... which satisfies dependency `xtensa-lx-rt = "^0.16.0"` (locked to 0.16.0) of package `esp-hal v0.19.0 (/esp-hal/esp-hal)`
```

### PR DESCRIPTION
**Pull request by jessebraham**
~~Still an Xtensa-related issue to iron out in `esp-wifi` but @bugadani will take care of that. Once that is resolved CI should be green across the board.~~

Otherwise, things should be in order as far as I can tell. As always though, would appreciate a handful of reviews here to make sure I am not missing anything.

The following packages will ultimately be published:

- `esp-backtrace`
- `esp-hal`
- `esp-hal-embassy`
- `esp-hal-procmacros`
- `esp-hal-smartled`
- `esp-ieee802154`
- `esp-metadata`
- `esp-println`
- `esp-wifi`
- `xtensa-lx-rt`

Closes #1890 
Closes #1897

### CODE REVIEW

**Comment by jessebraham** [comment]
We're not publishing `esp-lp-hal` and I didn't bump its version, so not sure why the changelog check is failing there?

**Comment by jessebraham** [comment]
Changelog check was failing on `esp-lp-hal`, which is not being published, so I've added the `skip-changelog` label just to get that green.

**Comment by bugadani** [comment]
I guess the Cargo.toml change in esp-lp-hal triggered the changelog check

File: xtensa-lx-rt/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name         = "xtensa-lx-rt"
-version      = "0.16.0"
+version      = "0.16.1"
```
Comment (bjoernQ) [inline xtensa-lx-rt/Cargo.toml:3]: Given the update breaks esp-wifi shouldn't this be `0.17.0`?

Comment (bugadani) [reply xtensa-lx-rt/Cargo.toml:3]: Yeah I agree

Comment (jessebraham) [reply xtensa-lx-rt/Cargo.toml:3]: Yeah wasn't aware of the break when I changed that, thanks

**Review by bjoernQ — APPROVED** [review]
Thanks!

**Review by SergioGasquez — APPROVED** [review]
LGTM! Thanks

**Review by MabezDev — APPROVED** [review]
Thanks!

**Pull request merged**